src/main/java/bisq/core/offer/OfferUtil.java

```diff
@@ -137,52 +142,86 @@ public static boolean isBsqForFeeAvailable(BsqWalletService bsqWalletService, @N
+    /**
+     *
+     * @param volumeByAmount      The volume generated from an amount
+     * @param factor              The factor used for rounding. E.g. 1 means rounded to units of 1 EUR, 10 means rounded to 10 EUR...
+     * @return The adjusted Fiat volume
+     */
+    @VisibleForTesting
+    static Volume getAdjustedFiatVolume(Volume volumeByAmount, int factor) {
+        // Fiat currencies use precision 4 and we want multiple of factor so we divide by 10000 * factor then
         // round and multiply with factor
-        long rounded = Math.max(1, Math.round((double) volumeByAmount.getValue() / (10000d * factor)));
-        // We pass EUR but any fiat currency would be good.
-        return Volume.parse(String.valueOf(rounded * factor), currencyCode);
+        long roundedVolume = Math.round((double) volumeByAmount.getValue() / (10000d * factor)) * factor;
+        // Smallest allowed volume is factor (e.g. 10 EUR or 1 EUR,...)
+        roundedVolume = Math.max(factor, roundedVolume);
+        return Volume.parse(String.valueOf(roundedVolume), volumeByAmount.getCurrencyCode());
     }
 
-    public static Coin getRoundedFiatAmount(Coin amount, Price price, String currencyCode, long maxTradeLimit) {
-        return getAdjustedAmount(amount, price, maxTradeLimit, currencyCode, 1);
+    public static Coin getRoundedFiatAmount(Coin amount, Price price, long maxTradeLimit) {
+        return getAdjustedAmount(amount, price, maxTradeLimit, 1);
     }
 
     public static Coin getAdjustedAmountForHalCash(Coin amount, Price price, long maxTradeLimit) {
-        return getAdjustedAmount(amount, price, maxTradeLimit, "EUR", 10);
+        return getAdjustedAmount(amount, price, maxTradeLimit, 10);
     }
 
-    private static Coin getAdjustedAmount(Coin amount, Price price, long maxTradeLimit, String currencyCode, int factor) {
-        // Amount must result in a volume of min factor units of the fiat currency, e.g. 1 EUR or 10 EUR in case of Halcash
-        Volume volumeRoundedToFactor = Volume.parse(String.valueOf(factor), currencyCode);
-        Coin amountByVolumeRoundedToFactor = price.getAmountByVolume(volumeRoundedToFactor);
-        // We set min amount so it has a volume of 10 EUR
-        if (amount.compareTo(amountByVolumeRoundedToFactor) < 0)
-            amount = amountByVolumeRoundedToFactor;
+    /**
+     *
+     * @param amount            Bitcoin amount which is a candidate for getting rounded
+     * @param price             Price used in relation ot that amount
+     * @param maxTradeLimit     The max. trade limit of the users account.
+     * @param factor            The factor used for rounding. E.g. 1 means rounded to units of 1 EUR, 10 means rounded to 10 EUR...
+     * @return The adjusted amount
+     */
+    @VisibleForTesting
+    static Coin getAdjustedAmount(Coin amount, Price price, long maxTradeLimit, int factor) {
```

Comment:
This method is fairly complex! It would be great if there were tests for it..

I have been working on that (and tests for some of the methods in `Coin`) which I will send out for review after this change goes in. I think I understand the behavior of the method, and that it's going to do what you want under normal conditions in terms of making user-provided values from the UI match the expected constraints.

Reply:
For future reference, those tests I mentioned above were merged in #160.